### PR TITLE
[Testonly] Extra BTC fee handling tests

### PR DESCRIPTION
**Pull request by catenocrypt**
## Description

Add extra tests for BTC fee estimation / UTXO selection.  Tests only, no prod code change.

## Testing instructions

Standard unit tests.

## Types of changes

* New tests

## Checklist

- [x] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.
- [x] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain
- [x] If there is a related Issue, mention it in the description (e.g. Fixes #<issue_number> ).

### CODE REVIEW

**Comment by catenocrypt** [comment]
Diff coverage metric is wrong.  This PR only adds tests, so coverage _can not_ decrease.

File: tests/Bitcoin/TransactionPlanTests.cpp [hunk 1]
```diff
@@ -173,6 +193,115 @@ TEST(TransactionPlan, UnpsentsInsufficient) {
...
+    //  3-in-2-out {600, 800, 1'000}     avail 2400 txamount 1775 fee 566 change 59
+    auto utxos = buildTestUTXOs({400, 500, 600, 800, 1'000});
+    auto byteFee = 2;
+    auto sigingInput = buildSigningInput(1'775, byteFee, utxos);
+
+    auto txPlan = TransactionBuilder::plan(sigingInput);
+
+    EXPECT_TRUE(verifyPlan(txPlan, {}, 0, 0));
+    //EXPECT_TRUE(verifyPlan(txPlan, {500, 600, 800, 1'000}, 1'775, 702));
+    //EXPECT_TRUE(verifyPlan(txPlan, {600, 800, 1'000}, 1'775, 566));
```
Comment (hewigovens) [inline tests/Bitcoin/TransactionPlanTests.cpp:231]:
```suggestion
```

**Review by hewigovens — COMMENTED** [review]
can you check coverage setup? it doesn't show files changed, looks like always full diff

**Review by hewigovens — APPROVED** [review]

**Comment by hewigovens** [comment]
I create an issue for codecov: https://github.com/trustwallet/wallet-core/issues/1239 can you take a look later?

**Review by vcoolish — APPROVED** [review]

**Review by vikmeup — APPROVED** [review]

**Pull request merged**